src/poc/miner_onion_server_light.erl

```diff
@@ -325,18 +330,31 @@ try_decrypt(IV, OnionCompactKey, _OnionKeyHash, Tag, CipherText, ECDHFun) ->
 %%    end.
 
 -spec tx_power(Region :: atom(), State :: state()) -> {ok, pos_integer(), pos_integer(), non_neg_integer()} | {error, any()}.
-tx_power(Region, #state{compact_key=_CK, region_params = RegionParams}) ->
+tx_power(Region, #state{compact_key=_CK, region_params = RegionParams, gain = AssertGain}) ->
     try
         MaxEIRP = lists:max([blockchain_region_param_v1:max_eirp(R) || R <- RegionParams]),
         %% if the antenna gain is accounted for in the packet forwarder config file
         %% set this to false
-        %% ConsiderTxGain = application:get_env(miner, consider_tx_gain, true),
-        %% TODO - revisit as we are dropping the GW gain from the ledger
-        %%        do we need an API to pull this from a validator ?
-        EIRP = trunc(MaxEIRP/10),
-        lager:info("Region: ~p, Gain: ~p, MaxEIRP: ~p, EIRP: ~p",
-                   [Region, undefined, MaxEIRP/10, EIRP]),
-        {ok, EIRP, EIRP, 0}
+        ConsiderTxGain = application:get_env(miner, consider_tx_gain, true),
+        case AssertGain of
+            X when X == undefined orelse ConsiderTxGain == false ->
+                %% No gain on chain or the gain is accounted for in pkt forwarder,
+                %% EIRP = max allowed in region
+                EIRP = trunc(MaxEIRP/10),
+                lager:info("Region: ~p, Gain: ~p, MaxEIRP: ~p, EIRP: ~p",
+                           [Region, undefined, MaxEIRP/10, EIRP]),
+                FinalGain = case AssertGain of
+                           undefined -> 0;
+                           _ -> AssertGain
```

Comment:
Does this need to be `trunc(AsserrGain/10)`?

Reply:
I followed same as the non light version ie https://github.com/helium/miner/blob/master/src/poc/miner_onion_server.erl#L502